admin/services/data-access/pupil-restart.data.service.js

```diff
@@ -46,10 +54,100 @@ pupilRestartDataService.findLatest = async function (options) {
 
 /**
  * Get all the restart codes documents
+ * @deprecated Please use sqlFindRestartCodes instead
  * @return {Promise.<{Object}>}
  */
 pupilRestartDataService.getRestartCodes = async () => {
   return RestartCode.find().lean().exec()
 }
 
+/** SQL METHODS **/
+
+/**
+ * Create a new pupil restart.
+ * @param {object} data
+ * @return  { insertId: <number>, rowsModified: <number> }
+ */
+pupilRestartDataService.sqlCreate = async (data) => {
+  return sqlService.create(table, data)
+}
+
+/**
+ * Returns number of restarts specified by pupil id
+ * @param pupilId
+ * @return {Promise.<*>}
+ */
+pupilRestartDataService.sqlGetNumberOfRestartsByPupil = async function (pupilId) {
+  const sql = `SELECT COUNT(*) FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId`
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
+/**
+ * Find latest check for pupil
+ * @param pupilId - the pupil taking the check
+ * @return {Promise.<void>} - lean Check objects
+ */
+pupilRestartDataService.sqlFindLatestRestart = async function (pupilId) {
+  const sql = `SELECT TOP 1 * FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId AND isDeleted=0 ORDER BY createdAt DESC`
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    }
+  ]
+  const result = await sqlService.query(sql, params)
+  return R.head(result)
+}
+
+/**
+ * Find latest check for pupil
+ * @param pupilId - the pupil taking the check
+ * @return {Promise.<void>} - lean Check objects
+ */
+pupilRestartDataService.sqlFindRestartCodes = async function () {
+  const sql = `
+  SELECT 
+    id, 
+    code, 
+    statusDesc 
+  FROM ${sqlService.adminSchema}.[pupilRestartCode]
+  ORDER BY statusDesc ASC`
+  return sqlService.query(sql)
+}
+
+/**
+ * Mark an existing pupil restart as deleted
+ * @param pupilId
+ * @param userId
+ * @return {Promise<*>}
+ */
+pupilRestartDataService.sqlMarkRestartAsDeleted = async (pupilId, userId) => {
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    },
+    {
+      name: 'deletedByUser_id',
+      value: userId,
+      type: TYPES.Int
+    },
+    {
+      name: 'updatedAt',
```

Comment:
I don't think we should be manually updating the `updatedAt` column. @GuyHarwood - what are your thoughts?

Essentially `createdAt` and `updatedAt` are not application columns and can be thought to be owned by the Database.  We should not read or write to them.

If there is a business need for an updated-type column then another column should be added in.

Triggers will be added that update all the updatedAt columns automatically.

Reply:
Good point.  I overlooked this yesterday when George was asking.
For now - just don't update the field.  simply set isDeleted flag.  The audit process will take care of everything else when we implement post feb trial